x-pack/plugins/ml/public/application/notifications/components/ml_entity_selector.tsx

```diff
@@ -0,0 +1,186 @@
+            options: dfa.data_frame_analytics.map((v) => ({
+              label: v.id,
+              value: v.id,
+              key: `data_frame_analytics:${v.id}`,
+              color: visColorsBehindText[2],
+            })),
+          });
+        }
+      }
+
+      if (entityTypes?.trained_models) {
+        const models = await trainedModels.getTrainedModels();
+        if (models.length > 0) {
+          newOptions.push({
+            label: i18n.translate('xpack.ml.mlEntitySelector.trainedModelsLabel', {
+              defaultMessage: 'Trained models',
+            }),
+            isGroupLabelOption: true,
+            options: models.map((v) => ({
+              label: v.model_id,
+              value: v.model_id,
+              key: `trained_models:${v.model_id}`,
+              color: visColorsBehindText[3],
+            })),
+          });
+        }
+      }
+
+      setOptions(newOptions);
+    } catch (error) {
+      displayErrorToast(
+        error,
+        i18n.translate('xpack.ml.mlEntitySelector.fetchError', {
+          defaultMessage: 'Failed to fetch ML entities',
+        })
+      );
+    }
+    setIsLoading(false);
+  }, [
+    jobsApi,
+    trainedModels,
+    dataFrameAnalytics,
+    entityTypes,
+    visColorsBehindText,
+    displayErrorToast,
+  ]);
+
+  const selectedEntities = useMemo<Array<EuiComboBoxOptionOption<string>>>(() => {
+    return (selectedOptions ?? []).flatMap((o) => {
+      const fromOptions = options
+        .flatMap((g) => g.options)
+        .filter((op): op is EuiComboBoxOptionOption<string> => op!.value === o.id);
+      return fromOptions.length > 0
+        ? fromOptions
+        : [{ value: o.id, label: o.id, key: `unknown:${o.id}` }];
+    });
+  }, [options, selectedOptions]);
+
+  useEffect(
+    function fetchOptionsOnMount() {
+      fetchOptions();
+    },
+    [fetchOptions]
+  );
+
+  const onChange = useCallback<Exclude<EuiComboBoxProps<string>['onChange'], undefined>>(
+    (selection) => {
+      if (!onSelectionChange) return;
+
+      let resultSelection = selection;
+
+      if (handleDuplicates) {
+        const prevCounts = countBy(selectedEntities, 'value');
+        const currentCounts = countBy(selection, 'value');
+        resultSelection = resultSelection.filter(({ value }) => {
+          // If an ID with duplicates has been removed, delete all of them.
+          return !(prevCounts[value!] > 1 && currentCounts[value!] < prevCounts[value!]);
+        });
+      }
+
+      onSelectionChange(
+        resultSelection.map((s) => {
+          const [type] = s.key!.split(':');
+          return { id: s.value!, type: type as EntityType };
+        })
+      );
+    },
+    [onSelectionChange, selectedEntities, handleDuplicates]
+  );
+
+  return (
+    <EuiComboBox<string>
+      autoFocus={true}
+      isLoading={isLoading}
+      singleSelection={!multiSelect}
+      selectedOptions={selectedEntities}
+      options={options}
```

Comment:
Entities that had notifications, but have since been deleted, cannot be selected. Is there a way to populate it with all entities that have had notifications? 

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/7405507/195046447-58dcd9a1-610c-4bdd-8982-f9a7494248cd.png">

Reply:
Note that I hadn't synchronized the saved objects for the `gallery_delete` job here. Therefore the old saved object for this job still existed. Do you think we should be showing notifications for a deleted job in this case @jgowdyelastic ?

Reply:
Not sure it's possible, because of the space awareness. It'd require an extra API call to retrieve all unique terms for `job_id` from `.ml-notifications*`, but eventually we won't be able to resolve what IDs are available for the user.

Reply:
I think the alternative is too tricky to implement. We'd have to check every job to see whether it does still exist.
Just to clarify what's happening here...
When a job is deleted, it happens in the background in es, we do not know when the job has actually been deleted so we can't be sure when to delete the saved object.
It was decided that leaving these orphaned saved objects behind was not a problem. They are deleted on sync or when a new job arrives with the same name.

The reason `gallery_delete` is showing up here is because the saved object still exists.
IMO the behaviour here is ok. It's not ideal, but the alternative of checking whether a job really still exists before listing its messages would have quite an overhead.